alma/views/js/alma-product-insurance.js

```diff
@@ -206,92 +226,101 @@
-            inputsInsurance.forEach((input) => {
-                input.remove();
-            });
+            removeInsuranceInputs();
         }
 
         function addModalListenerToAddToCart() {
             if (AlmaInsurance.settings.isAddToCartPopupActivated === true && almaEligibilityAnswer) {
                 const addToCart = getAddToCartButton();
                 if (addToCart) {
                     // If we change the quantity the DOM is reloaded then we need to remove and add the listener again
-                    addToCart.removeEventListener("click", insuranceListener);
-                    addToCart.addEventListener("click", insuranceListener);
+                    addToCart.removeEventListener("click", addToCartListener);
+                    addToCart.addEventListener("click", addToCartListener);
                 }
             }
         }
 
-        function getAddToCartButton() {
-            let addToCart = document.querySelector('button.add-to-cart');
-            // TODO: Ravate specific to generalise with selector configuration
-            if (!addToCart) {
-                addToCart = document.querySelector('.add-to-cart a, .add-to-cart button').first();
-            }
+        // Find the selector for the add to cart button
+        function getAddToCartBtnSelector() {
+            // TODO: move selectors to a module configuration option
+            const selectors = [
+                '[data-button-action=add-to-cart]:visible', // generic PrestaShop add to cart button
+                'button.add-to-cart:visible',
+                '.add-to-cart a:visible',
+                '.add-to-cart button:visible',
+                'a[role=button][href$=addToCart]:visible' // elementor addToCart widget
+            ];
+
+            // Return first selector that successfully matches an element in the DOM
+            return selectors.find(selector => $(selector).length > 0)
+        }
 
-            return addToCart;
+        function getAddToCartButton() {
+            return $(getAddToCartBtnSelector())[0];
         }
 
-        function insuranceListener(event) {
-            if (!insuranceSelected) {
+        function addToCartListener(event) {
+            if (!insuranceSelected && !addToCartFlow) {
                 event.preventDefault();
                 event.stopPropagation();
-                openModal('popupModal', quantity);
-                insuranceSelected = true;
+
                 addToCartFlow = true;
+                openModal('popupModal', quantity);
+            } else {
+                addToCartFlow = false;
             }
-            insuranceSelected = false;
         }
 
+        // This is only used to display a callout message when the product page for the actual insurance "product" from
+        // the catalog is being viewed by a customer
+        // TODO: This should probably be removed in favor of conditional templating/hooks on the product page itself
```

Comment:
I agree, In my memory, I had analysed it but I didn't find any hook to display the message in this position

Reply:
Thx, I just added it as a note for exploration later :)

I think we could have all the conditions checks in PHP and use the generic header hook on the product page when the checks are true.
But again: nothing too urgent ^^